eternalcode-commons-folia/src/main/java/com/eternalcode/commons/folia/scheduler/FoliaTaskImpl.java

```diff
@@ -0,0 +1,28 @@
+package com.eternalcode.commons.folia.scheduler;
+
+import com.eternalcode.commons.scheduler.Task;
+import io.papermc.paper.threadedregions.scheduler.ScheduledTask;
+
+public class FoliaTaskImpl implements Task {
+
+    private final ScheduledTask rootTask;
+
+    public FoliaTaskImpl(ScheduledTask rootTask) {
+        this.rootTask = rootTask;
+    }
+
+    @Override
+    public void cancel() {
+        this.rootTask.cancel();
+    }
+
+    @Override
+    public boolean isCanceled() {
+        return this.rootTask.isCancelled();
+    }
+
+    @Override
+    public boolean isAsync() {
+        return true;
```

Comment:
nie zawsze to jest async, ale w sumie ciężko powiedzieć co oznacza to już w tym kontekście bo jest po prostu multithreading. Można dać false jak jest global region scheduler

Reply:
Ogólnie, to w zasadzie nigdy nie wiadomo co tu dać, i global region scheduler też będzie tutaj niepoprawny, więc stwierdziłem że zawsze będę zwracać true